lib/galaxy/datatypes/sequence.py

```diff
@@ -1208,3 +1209,67 @@ def sniff(self, filename):
             pass
 
         return False
+
+
+class MemePsp(Sequence):
+    """Class representing MEME Position Specific Priors"""
+    file_ext = "memepsp"
+
+    def sniff( self, filename ):
+        """
+        The format of an entry in a PSP file is:
+
+        >ID WIDTH
+        PRIORS
+
+        For complete details see http://meme-suite.org/doc/psp-format.html
+
+        >>> from galaxy.datatypes.sniff import get_test_fname
+        >>> fname = get_test_fname('1.memepsp')
+        >>> MemePsp().sniff(fname)
+        True
+        >>> fname = get_test_fname('sequence.fasta')
+        >>> MemePsp().sniff(fname)
+        False
+        """
+        def floats_verified(l):
+            for item in l.split():
+                try:
+                    float(item)
+                except:
+                    return False
+            return True
+        try:
+            num_lines = 0
+            with open(filename) as fh:
+                line = fh.readline()
```

Comment:
Please limit these readlines in future - this will use an unbounded amount of memory on certain inputs.